### PR TITLE
Add custom user-agent header to v5 client requests

### PR DESCRIPTION
**Pull request by Tahnan**

### CODE REVIEW

File: luminoso_api/v5_client.py [hunk 1]
```diff
@@ -170,6 +178,11 @@ def _request(self, req_type, url, **kwargs):
         Make a request via the `requests` module. If the result has an HTTP
         error status, convert that to a Python exception.
         """
+        user_agent = 'LuminosoClient/' + VERSION
+        if self._user_agent_suffix is not None:
```
Comment (slibs63) [inline luminoso_api/v5_client.py]: Is there a reason this is being calculated for each request?  Why not just store `self.user_agent` at instantiation?

Comment (Tahnan) [reply luminoso_api/v5_client.py]: This is a fine question, and in fact I very much wanted to do this.  The problem is what happens when you use `client_for_path` and you want the new client to have the same suffix.  If you've stored the user agent as `LuminosoClient/2.0 lumi-upload` and pass that as the agent suffix, you'll get a new client with the user agent `LuminosoClient/2.0 LuminosoClient/2.0 lumi-upload`.

The alternatives I could see (optionally pass either a user-agent suffix *or* an entire user agent, to two different keyword arguments; when making a new client, split the user agent at the first space and send the rest of it as the suffix to the new client....) seemed dicey at best.  Having to do a very small calculation on each call whose time would be negligable compared to waiting for an API response seemed the best way to go.

Needless to say, open to suggestions.

Comment (slibs63) [reply luminoso_api/v5_client.py]: Couldn't you continue to pass the suffix, and then in the init method set it on self.user_agent?  As in:
```
def __init__(self, ..., user_agent_suffix=None):
    self.user_agent = 'LuminosoClient/' + VERSION
    self.user_agent_suffix = user_agent_suffix
    if user_agent_suffix is not None:
        self.user_agent += user_agent_suffix
```

Comment (Tahnan) [reply luminoso_api/v5_client.py]: What, store *both*?  That's just madness!

...I'll go fix that.

**Comment by slibs63** [comment]
I notice that the Trello card says this work is "to consider".  Are we still considering its merits, or are we all convinced?

File: CHANGELOG [hunk 2]
```diff
@@ -22,6 +22,10 @@ Version 2.0 (2019-07-03)
     if a domain with "daylight" was specified but no token was found, to help
     users transition to the new URLs.
 
+  * Added a custom "user-agent" to the v5 client's headers, along with
+    functionality to allow users (and specifically the CLI scripts) to add a
+    suffix to the user-agent
```
Comment (slibs63) [inline CHANGELOG]: This strikes me as a mostly internal mechanism.  Do we really want to mention it in the changelog?

Comment (Tahnan) [reply CHANGELOG]: I don't know.  It's external-facing functionality, and I think of changelogs as things that do include internal changes (in a sort of "no surprises, we're not trying to hide anything from you" sort of way).  Feel free to get other opinions though.

Comment (slibs63) [reply CHANGELOG]: I guess I just didn't know what the standard practice was for reporting on internal behavior in an external package.  I'm willing to believe that it belongs in the changelog.

**Pull request merged**